tests/python/test_comm_hooks_fsdp.py

```diff
@@ -399,6 +401,174 @@ def test_slowmo_optimizer_buffer(self, sharding_strategy):
         for entry in slowmo_statedict["state"].values():
             self.assertIn("slow_momentum", entry)
 
+    @skip_if_lt_x_gpu(2)
+    @parametrize("sharding_strategy", [ShardingStrategy.NO_SHARD])
+    def test_gossip_grad_state_init(self, sharding_strategy):
+        num_devices = torch.cuda.device_count()
+        with self.assertRaisesRegex(
+            ValueError,
+            "`local_process_group` and `num_nodes` should be provided together.",
+        ):
+            state = GossipGraDState(num_nodes=2)
+            state = GossipGraDState(
```

Comment:
This won't be tested I think. The context manager will bail out after L412 throws?

Reply:
I see, I'll adjust this